mitosheet/src/components/taskpanes/CodeOptions/CodeOptionsTaskpane.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useState } from "react";
+import MitoAPI from "../../../jupyter/api";
+import { AnalysisData, UIState, UserProfile } from "../../../types";
+
+import Toggle from "../../elements/Toggle";
+import DefaultTaskpane from "../DefaultTaskpane/DefaultTaskpane";
+import DefaultTaskpaneBody from "../DefaultTaskpane/DefaultTaskpaneBody";
+import DefaultTaskpaneHeader from "../DefaultTaskpane/DefaultTaskpaneHeader";
+import Row from "../../layout/Row";
+import Col from "../../layout/Col";
+import LabelAndTooltip from "../../elements/LabelAndTooltip";
+import Input from "../../elements/Input";
+import { useDebouncedEffect } from "../../../hooks/useDebouncedEffect";
+
+
+interface CodeOptionsTaskpaneProps {
+    mitoAPI: MitoAPI;
+    userProfile: UserProfile;
+    setUIState: React.Dispatch<React.SetStateAction<UIState>>;
+    analysisData: AnalysisData;
+}
+
+
+
+/* 
+    This is the CodeOptions taskpane, allows you to configure how the code is generated
+*/
+const CodeOptionsTaskpane = (props: CodeOptionsTaskpaneProps): JSX.Element => {
+
+    const [codeOptions, setCodeOptions] = useState(() => props.analysisData.codeOptions);
+
+    useDebouncedEffect(() => {
+        void props.mitoAPI.updateCodeOptions(codeOptions);
+    }, [codeOptions], 1000);
+
+    return (
+        <DefaultTaskpane>
+            <DefaultTaskpaneHeader 
+                header="Generated Code Options"
+                setUIState={props.setUIState} 
+            />
+            <DefaultTaskpaneBody
+                requiresEnterprise={{
+                    featureName: "code_options",
+                    mitoAPI: props.mitoAPI
+                }}
+                userProfile={props.userProfile}
+            >
+                <Row justify='space-between' align='center'>
+                    <Col>
+                        <LabelAndTooltip tooltip="A Python function is a reusable block of code that performs a specific task. It takes input, processes it, and returns output, making your code more organized and able to be easily rerun on new datasets.">
```

Comment:
When we do the release, let's make a how-to for this feature. We could link the the funciton explanation blog too